src/LanguageServer/Impl/Implementation/Server.cs

```diff
@@ -269,19 +268,22 @@ public sealed partial class Server : IDisposable {
         public void NotifyPackagesChanged(CancellationToken cancellationToken = default) {
             var interpreter = _services.GetService<IPythonInterpreter>();
             _log?.Log(TraceEventType.Information, Resources.ReloadingModules);
-            // No need to reload typeshed resolution since it is a static storage.
-            // User does can add stubs while application is running, but it is
-            // by design at this time that the app should be restarted.
-            interpreter.ModuleResolution.ReloadAsync(cancellationToken).ContinueWith(t => {
-                _log?.Log(TraceEventType.Information, Resources.Done);
-                _log?.Log(TraceEventType.Information, Resources.AnalysisRestarted);
+
+            interpreter.ModuleResolution.BuiltinsModule.ResetAst();
+
+            interpreter.TypeshedResolution.ReloadAsync().ContinueWith(async t => {
+                await interpreter.ModuleResolution.ReloadAsync();
+
+                _services.GetService<PythonAnalyzer>().GCNextSession();
```

Comment:
`<IPythonAnalyzer>`

Reply:
I'll be moving things down into PythonAnalyzer instead, that way I don't have to modify IBuiltinModule and so on.

Reply:
This doesn't have to be public API, there is `PythonAnalyzer.ResetAnalyzer` method that is called later.